src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FindJavaObjectsStep.cs

```diff
@@ -0,0 +1,94 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Java.Interop.Tools.JavaCallableWrappers.Adapters;
+using Java.Interop.Tools.JavaCallableWrappers.CallableWrapperMembers;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using Xamarin.Android.Tasks;
+
+namespace MonoDroid.Tuner;
+
+/// <summary>
+/// Scans an assembly for JLOs that need JCWs generated and writes them to an XML file.
+/// </summary>
+public class FindJavaObjectsStep : BaseStep
+{
+	public string ApplicationJavaClass { get; set; } = "";
+
+	public bool ErrorOnCustomJavaObject { get; set; }
+
+	public bool UseMarshalMethods { get; set; }
+
+	public TaskLoggingHelper Log { get; set; }
+
+	public FindJavaObjectsStep (TaskLoggingHelper log) => Log = log;
+
+	public bool ProcessAssembly (AssemblyDefinition assembly, string destinationJLOXml)
+	{
+		var action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
+
+		if (action == AssemblyAction.Delete)
+			return false;
+
+		var types = ScanForJavaTypes (assembly);
+		var initial_count = types.Count;
+
+		// Filter out Java types we don't care about
+		types = types.Where (t => !t.IsInterface && !JavaTypeScanner.ShouldSkipJavaCallableWrapperGeneration (t, Context)).ToList ();
+
+		Log.LogDebugMessage ($"{assembly.Name.Name} - Found {initial_count} Java types, filtered to {types.Count}");
+
+		var wrappers = ConvertToCallableWrappers (types);
+		XmlExporter.Export (destinationJLOXml, wrappers, true);
```

Comment:
Can this use:
```csharp
using (var sw = MemoryStreamPool.Shared.CreateStreamWriter ())
       XmlExporter.Export (sw, wrappers, true);
       Files.CopyIfStreamChanged (sw.BaseStream, destinationJLOXml);
}
```

Reply:
I'm not sure I understand this "optimization".

`CopyIfStreamChanged` adds the following overhead:
- Read existing file from disk
- Hash existing file
- Hash stream contents

At best, it eliminates writing the file to disk.  While I admit I haven't tested it, I suspect the reading + hashing (+ maybe writing) is more expensive than simply always writing the file.

Reply:
This is used for writing `*.java` files, as we were trying to prevent `javac` and `d8/r8` from running again. This made it where assemblies could change, `*.java` files kept same timestamps.

It may not be needed here if some other slow target isn't using the `.xml` files as inputs.

Reply:
Went ahead and made this change even though we don't benefit from it _today_.  It will likely be useful in the future when `<GenerateJavaCallableWrappers>` is moved to an incremental target.

We also no longer need the timestamp of the `.jlo.xml` to always be updated as it is no longer listed as a `Output` of `_LinkAssembliesNoShrink`.